python/ray/data/_internal/plan.py

```diff
@@ -393,20 +405,96 @@ def has_computed_output(self) -> bool:
         )
 
 
+def _pack_args(
```

Comment:
This is packaging it into a flat positional arg list so any Ray object futures are properly resolved when these arguments are passed into a Ray task. We still can't pass a `FnParams` data class around as a required argument since we want Datasets operations and UDFs to be able to opt-in to receiving these passthrough args, as mentioned here: https://github.com/ray-project/ray/pull/25613#discussion_r893840918